pkg/daemon/daemon.go

```diff
@@ -881,8 +882,8 @@ func checkFileContentsAndMode(filePath, expectedContent string, mode os.FileMode
 		glog.Errorf("could not read file: %q, error: %v", filePath, err)
 		return false
 	}
-	if strings.Compare(string(contents), expectedContent) != 0 {
-		glog.Errorf("content mismatch for file: %q; expected: %v; received: %v", filePath, expectedContent, string(contents))
+	if !bytes.Equal(contents, expectedContent) {
+		glog.Errorf("content mismatch for file: %q; expected: %v; received: %v", filePath, expectedContent, contents)
```

Comment:
Hmm, this will print the contents as a byte array, right? We could not print the raw contents at all, though it's definitely helpful if it is indeed human-readable text (which it likely is in most cases). WDYT about casting to `string` here before printing and otherwise fallback on printing as a byte array?

I'm good with this as is too though and doing that as a follow up.

Reply:
the issue is casting to string a bytes slice always succeeds, not sure how to differentiate if we can't show something, I'm not really sure this info is helpful for big files as well :thinking:

Reply:
How about we just tweak it to `"content mismatch for file: %q"` for now?

Reply:
sounds great :+1:

Reply:
done